packages/taquito-rpc/src/taquito-rpc.ts

```diff
@@ -95,6 +101,18 @@ export class RpcClient implements RpcClientInterface {
     return `${this.url.replace(/\/+$/g, '')}${path}`;
   }
 
+  private validateAddress(address: string) {
+    if (validateAddress(address) !== ValidationResult.VALID) {
+      throw new InvalidAddressError(`Invalid address: ${address}`)
+    }
+  }
+
+  private validateContract(address: string) {
+    if (validateAddress(address) !== ValidationResult.VALID) {
```

Comment:
The validator used should be `validateContractAddress` instead of `validateAddress`.

Reply:
fixed